troposphere/serverless.py

```diff
@@ -31,6 +31,19 @@ def policy_validator(x):
                          + " policy documents")
 
 
+class DeadLetterQueue(AWSProperty):
+    props = {
+        'Type': (basestring, False),
+        'TargetArn': (basestring, False)
+    }
+
+    def validate(self):
+        valid_types = ['SQS', 'SNS']
+        if self.properties['Type'] not in valid_types:
```

Comment:
Type is not required so you need to check for existence before dereferencing it. 
```
if 'Type' in self.properties and \
    self.properties['Type'] not in valid_types:
...
```